src/frontend/staff-portal/src/app/components/staff-workbench/ticket-info/ticket-info.component.ts

```diff
@@ -702,19 +699,17 @@ export class TicketInfoComponent implements OnInit {
     let countForm = this.form.get('violationTicket').get('violationTicketCount' + countNo.toString());
     let parts = fullDescription.split(" "); // act/code fullsection description
 
-    // lookup legal statute from part[1] which should be in legal paragraph form
+    // lookup legal statute from part[0] which should be in legal paragraph form
     if (parts && parts.length > 1) {
-      let foundStatute = this.lookupsService.statutes?.filter(x => x.code === parts[1] && x.actCode === parts[0]).shift();
+      let foundStatute = this.lookupsService.statutes?.find(x => x.code === parts[0]);
```

Comment:
.find() is more performant than .filter()
Also removed filtering by parts[0], the ACT.

Reply:
Major difference is filter gives list of items whereas find gives you just 1 record (i.e., first matched record). It depends on the usecase you use.

Reply:
Right, but in this case, the code was just using shift(), which pulls the first item off the list. So before it was searching the entire list and then retrieving the first item. Now, it searches up to the first item, returns that and stops.

Reply:
That's good implementation.